File: server/src/main/java/xyz/chener/ext/napt/server/entity/ClientItem.java
package xyz.chener.ext.napt.server.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.activerecord.Model;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Date;


@EqualsAndHashCode(callSuper = false)
@TableName("client_item")
@Data
public class ClientItem extends Model<ClientItem> {

    @TableId(type = IdType.AUTO)
    private Integer id;

    private String clientUid;

    // 用于服务器请求的地址
    private String clientAddr;

    // 用于服务器被请求的端口
    private Integer serverPort;

    // 类型 tcp or  udp
    private Integer requestNtType;

    // 已使用流量 字节
    private Long flow;

    // 最大流量限制 字节   -1不限制
    private Long maxFlowLimit;

    // 速度限制  每秒多少字节  -1不限制
    private Long speedLimit;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private Date createTime;

    private String remark;



}

File: server/src/main/kotlin/xyz/chener/napt/server/entity/ClientItem.kt
package xyz.chener.napt.server.entity

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import jakarta.validation.constraints.NotNull
import java.util.*


@Entity
@Table(name = "client_item", indexes = [
    Index(name = "idx_client_uid", columnList = "client_uid", unique = false)
])
open class ClientItem {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    open var id: Long? = null

    @Column(name = "client_uid", length = 255)
    @NotNull
    open var clientUid: String? = null

    // 用于服务器请求的地址
    @Column(name = "client_addr", length = 255)
    @NotNull
    open var clientAddr: String? = null

    // 用于服务器被请求的端口
    @Column(name = "server_port")
    @NotNull
    open var serverPort: Int? = null

    // 类型 tcp or  udp
    @Column(name = "client_nt_type")
    @NotNull
    open var requestNtType: Int? = null

    // 已使用流量 字节
    @Column(name = "flow")
    @NotNull
    open var flow: Long? = null

    // 最大流量限制 字节   -1不限制
    @Column(name = "max_flow_limit")
    @NotNull
    open var maxFlowLimit: Long? = null

    // 速度限制  每秒多少字节  -1不限制
    @Column(name = "speed_limit")
    @NotNull
    open var speedLimit: Long? = null

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    @Column(name = "create_time")
    open var createTime: Date? = null

    @Column(name = "remark", length = 255)
    open var remark: String? = null


}